File: AtCoder/abc146/c/Main.java
import java.util.*;

class Main {
    final Scanner sc = new Scanner(System.in);

    public static void main(String[] args) {
        new Main().run();
    }

    long a;
    long b;
    long x;

    private void run() {
        a = sc.nextLong();
        b = sc.nextLong();
        x = sc.nextLong();

        long n = 1000000000;
        if (canBuy(n)) {
            System.out.println(n);
            return;
        }
        long left = 0;
        long right = n;
        long mid = 0;
        while (right - left > 1) {
            mid = (left + right) / 2;
            if (canBuy(mid)) {
                left = mid;
            } else {
                right = mid;
            }
        }
        System.out.println(left);
    }

    private boolean canBuy(long n) {
        return a * n + b * Long.toString(n).length() <= x;
    }
}

File: AtCoder/abc146/c/main.py
A, B, X = map(int, input().split())


def can_buy(n):
    return A * n + B * len(str(n)) <= X


ok = 0
ng = 10 ** 9 + 1

while ng - ok > 1:
    mid = (ok + ng) // 2
    if can_buy(mid):
        ok = mid
    else:
        ng = mid
print(ok)
